app/helpers/splash_screen_helper.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module SplashScreenHelper
+  def self.inline_splash_screen_script
+    <<~HTML.html_safe
+      <script>#{raw_js}</script>
+    HTML
+  end
+
+  def self.fingerprint
+    if Rails.env.development?
+      calculate_fingerprint
+    else
+      @fingerprint ||= calculate_fingerprint
+    end
+  end
+
+  private
+
+  def self.load_js
+    File.read("#{Rails.root}/app/assets/javascripts/discourse/dist/assets/splash-screen.js").sub("//# sourceMappingURL=splash-screen.map\n", "")
```

Comment:
When we finally move the frontend out of `app/assets/javascripts` we'll have to do a fair bit of `grep`-ing 😅